Multisig/Features/Security/SecurityCenter.swift

```diff
@@ -94,8 +94,65 @@ class SecurityCenter {
         }
     }
 
-    private static func migrateFromKeychainStorageToSecurityEnclave() {
-        //TODO:
+    func migrateFromKeychainStorageToSecurityEnclave() {
+        let task = { [unowned self] passcode in
+            do {
+                let keys = try KeyInfo.all()
+                // Migrate private keys
+                keys.compactMap { try? $0.privateKey() }.forEach {
+                    SecurityCenter.shared.import(id: DataID(id: $0.id), data: $0.keyData) { result in
+                        // TODO: Rollback on error
+                    }
+                }
+
+                // Migrate delegate keys
+                keys.compactMap { try? $0.delegatePrivateKey() }.forEach {
+                    SecurityCenter.shared.import(id: DataID(id: $0.id), data: $0.keyData, protectionClass: .data) { result in
+                        // TODO: Rollback on error
+                    }
+                }
+
+                // Migrate to new lock method
+                if App.shared.auth.isPasscodeSetAndAvailable {
```

Comment:
it will be very helpful to have unit tests for migrating different protection settings

Reply:
I don't have full imagination what should be in the unit test for this, if you have please share

Reply:
having different initial lock settings before migration ensuring that post-migration lock settings are correct. Also making sure that keys are migrated to a correct store. E.g. delegate keys to .data, private keys to .sensitive.